SpatialGDK/Source/SpatialGDK/Private/SpatialView/SubView.cpp

```diff
@@ -0,0 +1,209 @@
+	// that complete entities is always sorted. This would also need to be enforced in the temporarily incomplete case.
+	// If this sort shows up in a profile it would be worth trying.
+	Algo::Sort(CompleteEntities);
+	Algo::Sort(NewlyCompleteEntities);
+	Algo::Sort(NewlyIncompleteEntities);
+	Algo::Sort(TemporarilyIncompleteEntities);
+
+	Delta.Project(SubViewDelta, CompleteEntities, NewlyCompleteEntities, NewlyIncompleteEntities, TemporarilyIncompleteEntities);
+
+	CompleteEntities.Append(NewlyCompleteEntities);
+	NewlyCompleteEntities.Empty();
+	NewlyIncompleteEntities.Empty();
+	TemporarilyIncompleteEntities.Empty();
+}
+
+const FSubViewDelta& FSubView::GetViewDelta() const
+{
+	return SubViewDelta;
+}
+
+void FSubView::RefreshEntity(const Worker_EntityId& EntityId)
+{
+	if (TaggedEntities.Contains(EntityId))
+	{
+		CheckEntityAgainstFilter(EntityId);
+	}
+}
+
+FDispatcherRefreshCallback FSubView::CreateComponentExistenceRefreshCallback(FDispatcher& Dispatcher, const Worker_ComponentId& ComponentId,
+																			 const FComponentChangeRefreshPredicate& RefreshPredicate)
+{
+	return [ComponentId, &Dispatcher, RefreshPredicate](const FRefreshCallback& Callback) {
+		Dispatcher.RegisterComponentAddedCallback(ComponentId, [RefreshPredicate, Callback](const FEntityComponentChange& Change) {
+			if (RefreshPredicate(Change))
+			{
+				Callback(Change.EntityId);
+			}
+		});
+		Dispatcher.RegisterComponentRemovedCallback(ComponentId, [RefreshPredicate, Callback](const FEntityComponentChange& Change) {
+			if (RefreshPredicate(Change))
+			{
+				Callback(Change.EntityId);
+			}
+		});
+	};
+}
+
+FDispatcherRefreshCallback FSubView::CreateComponentChangedRefreshCallback(FDispatcher& Dispatcher, const Worker_ComponentId& ComponentId,
+																		   const FComponentChangeRefreshPredicate& RefreshPredicate)
+{
+	return [ComponentId, &Dispatcher, RefreshPredicate](const FRefreshCallback& Callback) {
+		Dispatcher.RegisterComponentValueCallback(ComponentId, [RefreshPredicate, Callback](const FEntityComponentChange Change) {
+			if (RefreshPredicate(Change))
+			{
+				Callback(Change.EntityId);
+			}
+		});
+	};
+}
+
+FDispatcherRefreshCallback FSubView::CreateAuthorityChangeRefreshCallback(FDispatcher& Dispatcher, const Worker_ComponentId& ComponentId,
+																		  const FAuthorityChangeRefreshPredicate& RefreshPredicate)
+{
+	return [ComponentId, &Dispatcher, RefreshPredicate](const FRefreshCallback& Callback) {
+		Dispatcher.RegisterAuthorityGainedCallback(ComponentId, [RefreshPredicate, Callback](const Worker_EntityId& Id) {
+			if (RefreshPredicate(Id))
+			{
+				Callback(Id);
+			}
+		});
+		Dispatcher.RegisterAuthorityLostCallback(ComponentId, [RefreshPredicate, Callback](const Worker_EntityId& Id) {
+			if (RefreshPredicate(Id))
+			{
+				Callback(Id);
+			}
+		});
+	};
+}
+
+void FSubView::RegisterTagCallbacks(FDispatcher& Dispatcher)
```

Comment:
Do we have strong guarantees that the Dispatcher will always outlive SubViews?

Reply:
yep! The dispatcher is tied to the lifetime of the worker, and subviews to the systems running on the worker.